pennylane/math/_is_independent.py

```diff
@@ -0,0 +1,317 @@
+    if interface == "tf":
+        import tensorflow as tf  # pylint: disable=import-outside-toplevel
+
+        tf.random.set_seed(seed)
+        rnd_args = []
+        for _ in range(num):
+            _args = (tf.random.uniform(tf.shape(_arg)) * width + bounds[0] for _arg in args)
+            _args = tuple(
+                tf.Variable(_arg) if isinstance(arg, tf.Variable) else _arg
+                for _arg, arg in zip(_args, args)
+            )
+            rnd_args.append(_args)
+    elif interface == "torch":
+        import torch  # pylint: disable=import-outside-toplevel
+
+        torch.random.manual_seed(seed)
+        rnd_args = [
+            tuple(torch.rand(np.shape(arg)) * width + bounds[0] for arg in args) for _ in range(num)
+        ]
+    else:
+        np.random.seed(seed)
+        rnd_args = [
+            tuple(np.random.random(np.shape(arg)) * width + bounds[0] for arg in args)
+            for _ in range(num)
+        ]
+
+    return rnd_args
+
+
+def _is_independent_num(func, interface, args, kwargs, test_kwargs):
+    """Test whether a function returns the same output at random positions.
+
+    Args:
+        func (callable): Function to be tested
+        interface (str): Interface used by ``func``
+        args (tuple): Positional arguments with respect to which to test
+        kwargs (dict): Keyword arguments for ``func`` at which to test;
+            The ``kwargs`` are kept fix in this test.
+        test_kwargs (dict): Options for the test.
+
+    Returns:
+        bool: Whether ``func`` returns the same output at the randomly
+        chosen points.
+
+    The available options via ``test_kwargs`` are
+
+      - ``num_pos=5``: Number of random positions to test
+      - ``seed=9123``: Seed for random number generator
+      - ``atol=1e-6``: Absolute precision for comparing the outputs
+      - ``rtol=0``: Absolute precision for comparing the outputs
+      - ``bounds=(-np.pi, np.pi)``: Limits of the range from which to sample
+
+    """
+    test_kwargs = test_kwargs or {}
+    num_pos = test_kwargs.get("num_pos", 5)
+    seed = test_kwargs.get("seed", 9123)
+    atol = test_kwargs.get("atol", 1e-6)
+    rtol = test_kwargs.get("rtol", 0)
+    bounds = test_kwargs.get("bounds", (-np.pi, np.pi))
+
+    rnd_args = _get_random_args(args, interface, num_pos, seed, bounds)
+    original_output = func(*args, **kwargs)
+    is_tuple_valued = isinstance(original_output, tuple)
+    for _rnd_args in rnd_args:
+        new_output = func(*_rnd_args, **kwargs)
+        if is_tuple_valued:
+            if not all(
+                np.allclose(new, orig, atol=atol, rtol=rtol)
+                for new, orig in zip(new_output, original_output)
+            ):
+                return False
+        else:
+            if not np.allclose(new_output, original_output, atol=atol, rtol=rtol):
+                return False
+
+    return True
+
+
+def _is_independent(func, interface, args, kwargs=None, num_kwargs=None):
```

Comment:
do you think it's worth having options to choose analytic, numerical, or both?

Reply:
Mhm. my spontaneous reaction is that this is UI overkill of something "almost non-public" :D 
I'm not sure, also if we unpack the `num_kwargs` arguments, this will become very large.
Any preference on your side? One could then raise a `NotImplementedError` in the `analytic, PyTorch` case.
The warning we currently have could instead be put into `spectrum`.

Reply:
This is minor on my side, so happy to go with your preferred solution here!